### ISSUE TITLE
misc/test_softmax.py::test_scale_mask_softmax_inplace mismatch

### ISSUE THREAD

**Issue by bbradelTT** [tenstorrent/tt-metal#14349]
Command to run:
pytest tests/tt_eager/python_api_testing/unit_testing/misc/test_softmax.py::test_scale_mask_softmax_inplace

**Comment by bbradelTT** [comment]
Disabling fp32_dest_acc_en allows the tests to have PCCs above 0.98
Although some tests 
Seems to be mostly the same as https://github.com/tenstorrent/tt-metal/issues/14352

cc @abhullar-tt @ncvetkovicTT

```
=============================================================== short test summary info ===============================================================
PASSED tests/tt_eager/python_api_testing/unit_testing/misc/test_softmax.py::test_scale_mask_softmax_inplace[float32-in0_DRAM-causal-384]
PASSED tests/tt_eager/python_api_testing/unit_testing/misc/test_softmax.py::test_scale_mask_softmax_inplace[float32-in0_DRAM-no-causal-384]
PASSED tests/tt_eager/python_api_testing/unit_testing/misc/test_softmax.py::test_scale_mask_softmax_inplace[bfloat16-in0_DRAM-causal-384]
PASSED tests/tt_eager/python_api_testing/unit_testing/misc/test_softmax.py::test_scale_mask_softmax_inplace[bfloat16-in0_DRAM-no-causal-384]
PASSED tests/tt_eager/python_api_testing/unit_testing/misc/test_softmax.py::test_scale_mask_softmax_inplace[bfloat8_b-in0_DRAM-causal-384]
PASSED tests/tt_eager/python_api_testing/unit_testing/misc/test_softmax.py::test_scale_mask_softmax_inplace[bfloat8_b-in0_DRAM-no-causal-384]
FAILED tests/tt_eager/python_api_testing/unit_testing/misc/test_softmax.py::test_scale_mask_softmax_inplace[float32-in0_DRAM-causal-64] - AssertionError: FAILED: 0.9898299995853843
FAILED tests/tt_eager/python_api_testing/unit_testing/misc/test_softmax.py::test_scale_mask_softmax_inplace[float32-in0_DRAM-no-causal-64] - AssertionError: FAILED: 0.9869795656618475
FAILED tests/tt_eager/python_api_testing/unit_testing/misc/test_softmax.py::test_scale_mask_softmax_inplace[bfloat16-in0_DRAM-causal-64] - AssertionError: FAILED: 0.9898318462118032
FAILED tests/tt_eager/python_api_testing/unit_testing/misc/test_softmax.py::test_scale_mask_softmax_inplace[bfloat16-in0_DRAM-no-causal-64] - AssertionError: FAILED: 0.986983253848556
FAILED tests/tt_eager/python_api_testing/unit_testing/misc/test_softmax.py::test_scale_mask_softmax_inplace[bfloat8_b-in0_DRAM-causal-64] - AssertionError: FAILED: 0.9897642164588555
FAILED tests/tt_eager/python_api_testing/unit_testing/misc/test_softmax.py::test_scale_mask_softmax_inplace[bfloat8_b-in0_DRAM-no-causal-64] - AssertionError: FAILED: 0.9869039339192907
================================================ 6 failed, 6 passed, 12 warnings in 138.60s (0:02:18) =================================================
```

Interestingly, that is also somehow causing non-float32 tests to fail and there is ND behaviour in the PCCs. A second run of one of the tests produced a different PCC:
```
=============================================================== short test summary info ===============================================================
FAILED tests/tt_eager/python_api_testing/unit_testing/misc/test_softmax.py::test_scale_mask_softmax_inplace[bfloat16-in0_DRAM-causal-64] - AssertionError: FAILED: 0.9898318462118032
============================================================ 1 failed, 1 warning in 13.84s ============================================================
```

**Comment by ncvetkovicTT** [comment]
@prajaramanTT @abhullar-tt @bbradelTT  Please note that previous comment isn't correct, #14352 and this issue don't have the same common cause.

**Comment by ncvetkovicTT** [comment]
@prajaramanTT @abhullar-tt @bbradelTT My bad, I rushed to comment all softmax and layernorm issues with the note that #14352 fix doesn't fix other issues, but it's not case with this one. 

Please see [ncvetkovic/14352_layernorm_fail](https://github.com/tenstorrent/tt-metal/tree/ncvetkovic/14352_layernorm_fail) and verify that all tests from ```test_softmax.py``` pass. If you're satisfied, we merge.

**Comment by bbradelTT** [comment]
After the small fix for https://github.com/tenstorrent/tt-metal/issues/15744 is in as well, then we can merge this in, yes.

**Comment by ncvetkovicTT** [comment]
Will be addressed as a part of [PR#15754](https://github.com/tenstorrent/tt-metal/pull/15754).

**Comment by ncvetkovicTT** [comment]
PR #15754 is merged, the issue can be closed.
